src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -73,14 +73,20 @@ public static ImmutableArray<Document> GetDocuments(this Solution solution, Uri
 
         public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution, Uri documentUri)
         {
-            // TODO: we need to normalize this. but for now, we check both absolute and local path
-            //       right now, based on who calls this, solution might has "/" or "\\" as directory
-            //       separator
+            // This logic needs to be cleaned up when we support URIs as a first class concept.
+            // For now we do our best to handle as many cases as we can.
+            // Tracking issue - https://github.com/dotnet/roslyn/issues/68083
+
             var documentIds = solution.GetDocumentIdsWithFilePath(documentUri.AbsolutePath);
             if (documentIds.Any())
                 return documentIds;
 
-            return solution.GetDocumentIdsWithFilePath(documentUri.LocalPath);
+            documentIds = solution.GetDocumentIdsWithFilePath(documentUri.LocalPath);
```

Comment:
How does AbsolutePath differ from LocalPath?

Reply:
Talked offline - absolute path is the path with potential query parameters (and possibly mismatching slashes).  Local path is supposed to better match the real file system path.  We've had problems in the past with only using one